src/cleanup-java.ts

```diff
@@ -14,4 +15,39 @@ async function run() {
   }
 }
 
-run();
+/**
+ * Check given input and run a save process for the specified package manager
+ * @returns Promise that will be resolved when the save process finishes
+ */
+async function saveCache() {
+  const cache = core.getInput(constants.INPUT_CACHE);
+  return cache ? save(cache) : Promise.resolve();
+}
+
+/**
+ * The save process is best-effort, and it should not make the workflow failed
+ * even though this process throws an error.
+ * @param promise the promise to ignore error from
+ * @returns Promise that will ignore error reported by the given promise
+ */
+async function ignoreError(promise: Promise<void>) {
+  return new Promise(resolve => {
+    promise
+      .catch(error => {
+        core.warning(error);
+        resolve(void 0);
+      })
+      .then(resolve);
+  });
+}
+
+export async function run() {
+  await Promise.all([removePrivateKeyFromKeychain(), ignoreError(saveCache())]);
```

Comment:
I would suggest to run both actions step by step:
```
await removePrivateKeyFromKeychain()
await ignoreError(saveCache())
```

If we run them in parallel, I guess the produced logs will be mixed that can complicate debugging. `removePrivateKeyFromKeychain` is pretty fast step so it is better to run it firstly.

Reply:
Fixed, thanks!